src/index.js

```diff
@@ -646,9 +787,14 @@ const handleDefaultRecords = (event, context, lintEnv) => {
 };
 
 const handleRecords = (event, context, lintEnv) => {
-  let source = event.Records[0].eventSource;
+  let source = [];
+  if (event.eventSource && event.eventSource === 'aws:kafka') {
+    source = event.eventSource;
+  }else{
+     source = event.Records[0].eventSource;
   if (event.Records[0].EventSource) {
     source = event.Records[0].EventSource;
+    }
```

Comment:
iam thinking should we create a seperate method say like this.

getEventSource(event)

inside this method we will do all the if else logics and add comments saying what it does.